inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.cpp

```diff
@@ -493,57 +496,90 @@ void MKLDNNConvolutionNode::initSupportedPrimitiveDescriptors() {
+                                                      inputDesc, weightDesc, outputDesc,
+                                                      mkldnn::memory::dims(stride.begin(), stride.end()),
+                                                      mkldnn::memory::dims(dilation.begin(), dilation.end()),
+                                                      mkldnn::memory::dims(paddingL.begin(), paddingL.end()),
+                                                      mkldnn::memory::dims(paddingR.begin(), paddingR.end())));
+    }
+    catch (...) {
+        IE_THROW() << "Cannot create convolution forward descriptor for layer: " << getName();
+    }
+    return conv_desc;
+}
+
+std::shared_ptr<mkldnn::convolution_forward::desc>
+MKLDNNConvolutionNode::createDescriptorInternal(const mkldnn::memory::desc& inputDesc,
+                                                const mkldnn::memory::desc& weightDesc,
+                                                const mkldnn::memory::desc& biasDesc,
+                                                const mkldnn::memory::desc& outputDesc,
+                                                mkldnn::algorithm alg) {
+    updatePadding();
+    std::shared_ptr<mkldnn::convolution_forward::desc> conv_desc;
+    try {
+        conv_desc.reset(new convolution_forward::desc(prop_kind::forward_scoring, alg,
+                                                      inputDesc, weightDesc, biasDesc, outputDesc,
+                                                      mkldnn::memory::dims(stride.begin(), stride.end()),
+                                                      mkldnn::memory::dims(dilation.begin(), dilation.end()),
+                                                      mkldnn::memory::dims(paddingL.begin(), paddingL.end()),
+                                                      mkldnn::memory::dims(paddingR.begin(), paddingR.end())));
+    }
+    catch (...) {
+        IE_THROW() << "Cannot create convolution forward descriptor for layer: " << getName();
+    }
+    return conv_desc;
+}
+
 void MKLDNNConvolutionNode::createDescriptor(const std::vector<MemoryDescPtr>& inputDesc,
                                              const std::vector<MemoryDescPtr>& outputDesc) {
-    const auto inDesc = MemoryDescUtils::convertToDnnlMemoryDesc(inputDesc[0])->getDnnlDesc();
-    const auto outDesc = MemoryDescUtils::convertToDnnlMemoryDesc(outputDesc[0])->getDnnlDesc();
+    auto inpDesc = inputDesc[0]->isDefined() ? inputDesc[0] : MemoryDescUtils::makeDummyDesc(*inputDesc[0]);
+    DnnlMemoryDescPtr definedInpMemDesc = MemoryDescUtils::convertToDnnlMemoryDesc(inpDesc);
+    DnnlMemoryDescPtr definedOutMemDesc;
+
+    if (outputDesc[0]->isDefined()) {
+        definedOutMemDesc = MemoryDescUtils::convertToDnnlMemoryDesc(outputDesc[0]);
+    } else {
+        std::vector<Shape> shapes = { definedInpMemDesc->getShape(), Shape(weightDims) };
+        auto outDims = shapeInferGeneric(shapes);
```

Comment:
also could you try to use new `shape_inference`?
if it will not work, create tickets on investigation and leave as is

Reply:
It was decided to adopt the new shape infer mechanism in separate ticket.